### ISSUE TITLE
Add ability to quote a value with delimiters in order to treat it as string

### ISSUE THREAD

**Issue by notxcain** [zio/zio-config#421]
The problem:
Given a config type `case class Config(foo: String, bar: List[String])` 
I want to be able to decode a `Map("foo"->"x,y,z", "bar" -> "a,b,c")` to `Config("x, y, z", "a" :: "b" :: "c" :: Nil)` 
Current implementation fails the decoding because encounters a `PropertyTree.Sequence` where `PropertyTree.Leaf` is expected.

The discussed (with @vigoo, on Discord) proposal is to add a `quote: Option[String]` and treat quoted values as strings.
e.g. If `quote = Some("'")` is set, then `Map("foo"->"'x,y,z'", "bar" -> "a,b,c")` decodes to `Config("x, y, z", "a" :: "b" :: "c" :: Nil)`

**Comment by afsalthaj** [comment]
Makes sense to have that distinction. Would you want to raise a PR on this?

**Comment by afsalthaj** [comment]
There is a way here

(string(“foo”) |@| string(“bar”).transformOrFail[List[String]](a => a.split(“,”).toList, _.mkString))(Config.apply, Config.unapply)


This is fair enough because there is theoretically no global definition of how a list is represented in a map-like Config source. Zio-Config allows u to have a global definition by making use of delimiters, however,  in the above example delimiter is being used in different context  within the same Config source, the above implementation makes sense, and in fact less magical and more readable. 


Quoting also makes sense, but the cost is less intuitive for a reader of your application. And accidental removal of quotes or having to explain why there is quote in some
Parts of the config. 

If you need this to be part of auto derivation, I would consider writing a new-type and define a Descriptor instance that does the same logic of converting a string to list. It’s a choice though. 


case class Config(foo: String, bar: Bar)

case class Bar(list: List[String]]

object Bar {
   implicit val descriptor: Descriptor[Bar] = Desriptor[String].transformOrFail(....)
}

descriptor[Config] will work as expected

**Comment by notxcain** [comment]
@afsalthaj 
After another round of thinking, I find it very confusing to add quoting and escaping as a part of the library UX. 

As an abstract client, I don't want to think about it at all. If I have `foo: String` I'm asking to parse a string, why do you try to parse something else? Delimeters should only be used for collection types. And should be a property of the collection decoder. Not the config source. 

I personally understand that there is an underlying `PropertyTree` representation, but once again it looks like parsing on demand would be a much better solution, and would also work for #418.
